source/simulator/solver.cc

```diff
@@ -936,7 +936,7 @@ namespace aspect
 
     // convert melt pressures:
     if (parameters.include_melt_transport)
-      melt_handler->compute_melt_variables(solution);
+      melt_handler->compute_melt_variables(system_matrix,solution,system_rhs);
```

Comment:
When you do this, we need to make sure that the system matrix will be reasssembled in the next step, right? Is this guaranteed to work? If this is already the case, we should leave a comment here.

Reply:
I do not quite understand your question. Why do we need to reassemble the matrix in the next step? We only use the fluid velocity blocks, all other parts of the matrix remain untouched.

Reply:
Oh, I thought we reuse the normal velocity block and we overwrite its content. Nevermind.